lib/IRGen/GenEnum.cpp

```diff
@@ -6107,6 +6105,15 @@ void IRGenModule::emitEnumDecl(EnumDecl *theEnum) {
   }
 
   emitFieldMetadataRecord(theEnum);
+
+  if (isResilient(theEnum, ResilienceExpansion::Maximal))
```

Comment:
Don't you want to return if its *not* resilient?

Reply:
The code below emits the global variables for resilient enum tags. I only want to proceed to this code if we are within the same resilience domain not outside.

Reply:
When are we ever resilient here? Isn’t this the code for emitting an enum _definition__? How can you define an enum that’s resilient to yourself?

Reply:
Good point. Rethinking that logic should we test if the enum is resilient and only then emit the globals (like you initially suggested)? Assuming, the transition from non-resilient to resilient is not allowed.

Reply:
Like so: https://github.com/apple/swift/pull/15401